internal/observability/events/options.go

```diff
@@ -0,0 +1,88 @@
+package event
+
+import (
+	"time"
+)
+
+// getOpts - iterate the inbound Options and return a struct.
+func getOpts(opt ...Option) options {
+	opts := getDefaultOptions()
+	for _, o := range opt {
+		o(&opts)
+	}
+	return opts
+}
+
+// Option - how Options are passed as arguments.
+type Option func(*options)
+
+// options = how options are represented
+type options struct {
+	withId          string
+	withDetails     map[string]interface{}
+	withHeader      map[string]interface{}
+	withFlush       bool
+	withRequestInfo *RequestInfo
+	withNow         time.Time
+	withRequest     *Request
+	withResponse    *Response
+	withAuth        *Auth
+}
+
+func getDefaultOptions() options {
+	return options{}
+}
+
+func WithId(id string) Option {
+	return func(o *options) {
+		o.withId = id
+	}
+}
+
+func WithDetails(d map[string]interface{}) Option {
+	return func(o *options) {
+		o.withDetails = d
+	}
+}
+
+func WithHeader(d map[string]interface{}) Option {
+	return func(o *options) {
+		o.withHeader = d
+	}
+}
+
+func WithFlush() Option {
+	return func(o *options) {
+		o.withFlush = true
+	}
+}
+
+func WithRequestInfo(i *RequestInfo) Option {
+	return func(o *options) {
+		o.withRequestInfo = i
+	}
+}
+
+func WithNow(now time.Time) Option {
```

Comment:
`event.WithNow` reads odd to me.

Reply:
+1 to this

Reply:
Hum... perhaps.  BTW, it's only used in tests, but had to be exported since some of those tests will not be in the event package but rather in event_test

Reply:
`event` is the package name, so not sure that can be changed.  The prefix `With` is our standard for options and `Now` aligns with the std lib `time.Now()` and every other instance of a time.Time that represents now.

Reply:
I think `WithTime` would be better? Given that it takes a `time.Time` and does not generate one via `time.Now()`.